Fix log-manager on Azure

vomba opened the pull request:
<!-- Choose your PR title carefully as it will be used as the entry in the changelog! -->

> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [ ] personal data beyond what is necessary for interacting with this pull request, nor
> - [ ] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       <!-- This PR adds a new feature -->
- [ ] kind/improvement   <!-- This PR changes an existing feature -->
- [ ] kind/deprecation   <!-- This PR removes an existing feature -->
- [ ] kind/documentation <!-- This PR contains documentation -->
- [ ] kind/clean-up      <!-- This PR cleans up technical debt -->
- [x] kind/bug           <!-- This PR fixes a bug -->
- [ ] kind/other         <!-- This PR does something else -->

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change <!-- This PR introduces an admin facing change, add "Platform Administrator notice" section -->
- [ ] kind/dev-change   <!-- This PR introduces a dev facing change, add "Application Developer notice" section -->
- [ ] kind/security     <!-- This PR introduces a critical security fix, add "Security notice" section -->
- [ ] [kind/adr]()      <!-- This PR implements an ADR, add the link -->

<!-- Uncomment the additional sections that applies. -->

<!-- Additional information to be added in the release notes
### Release notes
...
-->

<!-- Additional information with kind/admin-change
### Platform Administrator notice
...
-->

<!-- Add additional information with kind/dev-change
### Application Developer notice
...
-->

<!-- Add additional information with kind/security
### Security notice
...
-->

### What does this PR do / why do we need this PR?

<!-- Add description of the change -->
Updates the az cli commands and template files of log-manager.

<!-- Add all issues that are fixed by this PR, use "Part of" instead of "Fixes" if you want to keep issues open. -->
- Fixes #2239 

#### Information to reviewers

<!--
Any additional information reviews should know.

How to run / how to test.

Include screenshots if applicable to help explain these changes.
--->

#### Checklist

<!-- This section is not added to the changelog or release notes, it is to help you as a contributor and reviewers. -->

- [ ] Proper commit message prefix on all commits
  <!-- Example of commit message prefixes:
  - all: changes to multiple areas
  - apps: changes to applications running in all clusters
  - apps sc: changes to applications running in the service cluster
  - apps wc: changes to applications running in the workload cluster
  - bin: changes to management binaries or scripts
  - config: changes to configuration
  - docs: changes to documentation
  - release: release related
  - scripts: changes to other scripts
  - tests: changes to tests
  -->
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

aarnq commented:
Could you post a listing in ABS before and after a run, and the logs of the run for review please? :pray:

raviranjanelastisys reviewed (approved):
Nice work !! LGTM

aarnq commented:
> * I changed the azure plugin again, it successfully uploads gzipped chunks.
> 
> * LogManager compaction also successfully merges chunks, although it creates additional files somehow.

Check what the list indices function is doing, if it returns multiple entries for the same index.

vomba commented:
> > * I changed the azure plugin again, it successfully uploads gzipped chunks.
> > * LogManager compaction also successfully merges chunks, although it creates additional files somehow.
> 
> Check what the list indices function is doing, if it returns multiple entries for the same index.

Fixed it.

vomba commented on helmfile.d/values/fluentd/aggregator-common.yaml.gotmpl:
```diff
@@ -31,7 +31,7 @@ extraDeploy:
 image:
   registry: ghcr.io
   repository: elastisys/fluentd
-  tag: v5.19.0-ck8s1
+  tag: v5.19.1-ck8s1
```
I can add the Dockerfile, also it is good to note that [this](https://github.com/elastisys/fluentd-elasticsearch/) does not have the correct Dockerfile, we might need to remove referencing it.

vomba commented on config/sc-config.yaml:
```diff
@@ -1078,15 +1078,15 @@ fluentd:
         limits:
           cpu: 1000m
-          memory: 400Mi
+          memory: 2Gi
```
I believe so, because at some points when it's downloading/expanding chunks that are 500Mb or more it goes over 1Gb of memory. (It OOMed a lot before bumping that)

Zash replied:
Is there a way to control the size of those chunks? Or the compression level?

aarnq commented on config/providers/azure/sc-config.yaml:
```diff
@@ -2,3 +2,10 @@ harbor:
+  logManager:
+    compaction:
+      # azcopy is used by az CLI for downloading files
+      azcopyBufferGB: 0.3
+      azcopyConcurrency: 8
```
Could we type this out as `azureCopy` instead of `azcopy`? I think it'll make it more clear.
No one should really need to know that we are using `azcopy`, or rather that `az` cli uses `azcopy` in the background.

aarnq commented on config/schemas/config.yaml:
```diff
@@ -4974,6 +4974,12 @@ properties:
             properties:
+              azcopyBufferGB:
+                type: number
+                description: Memory buffer size in GB for azcopy CLI
```
```suggestion
                description: Configure the memory buffer size in GB for Azure copy operations.
```
And similarly for the concurrency configuration.

aarnq reviewed (commented):
Great find!

aarnq reviewed (approved):
LGTM
